### PR TITLE
Add Debian and README info describing config.yaml.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Adds debian as a checked platform and tries to add info about adding
further platforms to the README.

Debian Bullseye couldn't be added because there is not yet a bootstrap
repository for it.

### CODE REVIEW

File: test/rosdep_repo_check/README.md [hunk 1]
```diff
@@ -15,3 +15,13 @@ To review the entire rosdep database and not only the rules that were changed in
 
 PYTHONPATH=test python3 -m rosdep_repo_check
 ```
+
+## Adding new repository checks
+
+Platform checks can be added by updating [config.yaml](./config.yaml).
+
+* `package_sources` contains a set of repository base urls for each operating system distribution
+* `package_dashboards` contains an optional list of matching repository url patterns and template urls which can be used to extract and compose web links to packages in the matching distributions.
```
Comment (cottsay) [inline test/rosdep_repo_check/README.md:24]:
```suggestion
* `package_dashboards` contains an optional list of matching repository url patterns and template urls which can be used to extract and compose web links to packages in the matching distributions. This configuration is optional and may be omitted where appropriate.
```

File: test/rosdep_repo_check/README.md [hunk 2]
```diff
@@ -15,3 +15,13 @@ To review the entire rosdep database and not only the rules that were changed in
...
+* `supported_versions` lists of operating system versions or codenames that checks should be checked with the package sources
```
Comment (cottsay) [inline test/rosdep_repo_check/README.md:25]: Would you mind re-phrasing this?

File: test/rosdep_repo_check/README.md [hunk 3]
```diff
@@ -15,3 +15,13 @@ To review the entire rosdep database and not only the rules that were changed in
...
+* `supported_architectures` lists of operating system architectures that checks should be checked with the package sources. Although rosdep is expected to work across architectures repositories are only checked on amd64/x86_64 to save time. If a distribution has a radically different set of packages for different architectures checks for additional architectures can be added.
+
```
Comment (cottsay) [inline test/rosdep_repo_check/README.md:27]: Trailing newline.
```suggestion
```

File: test/rosdep_repo_check/README.md (hunk 2 above)
Comment (cottsay) [reply test/rosdep_repo_check/README.md:25]: Also, it might be worth noting that the last one on the list is used for trying to generate a 'suggestion'.

**Review by cottsay — APPROVED** [review]
A few suggestions, but overall, looks great!

File: test/rosdep_repo_check/README.md (hunk 2 above)
Comment (nuclearsandwich) [reply test/rosdep_repo_check/README.md]: Thanks for highlighting. That was definitely a case where I wrote the first and second halves of the sentence separately.

**Pull request merged**